app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,76 @@
+class TweetController < ApplicationController
+
+  ############ CREATE ###########
+  get '/tweets/new' do
+    if logged_in?(session)
+      erb :'tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] != ""
+      tweet = Tweet.new(params)
+      tweet.user_id = session[:id]
+      tweet.save
+      redirect to '/tweets'
+    end
+    redirect to '/tweets/new'
+  end
+
+  ############ READ ###########
+  get '/tweets' do
+    @tweets = Tweet.all
+    if logged_in?(session)
+      erb :'/tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?(session)
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  ############ UPDATE ###########
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by(id: params[:id])
+    if logged_in?(session) && User.current_user(session).id == @tweet.user_id
+      erb :'/tweets/edit_tweet'
+    elsif logged_in?(session) && User.current_user(session).id != @tweet.user_id
+      redirect to '/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+    if params[:content] != ""
+      @tweet.content = params[:content]
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...